src/components/movieDetails/MovieDetails.js

```diff
@@ -1,40 +1,71 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 import PropTypes from "prop-types";
 import "./MovieDetails.scss";
+import {useNavigate, useParams} from "react-router-dom";
+import convertDateIntoYear from "../../utils";
 
-const MovieDetails = ({movie}) => {
+const MovieDetails = () => {
+    const navigate = useNavigate();
+    const {movieId} = useParams();
+    const [movie, setMovie ] = useState();
+    useEffect(() => {
+        let controller, signal;
+        const fetchData = async () => {
+            if (controller) {
+                controller.abort()
+            }
+            controller = new AbortController();
+            signal = controller.signal;
+
+            const response = await fetch(`http://localhost:4000/movies/${movieId}`, {
+                signal: signal,
+            })
+            const data = await response.json();
+            console.warn("data", data)
+            controller = null;
+            setMovie(data);
+        }
+
+        fetchData();

```

Comment:
fetchData(movieId); and you can move the rest of method code from component to utils

Reply:
Done